### ISSUE TITLE
X-validate index error

### ISSUE THREAD

**Issue by blogeman** [DeepLabCut/DeepLabCut#989]
Windows 10, with an Anaconda Env, & DLC 2.2b8

I'm most comfortable using the GUI, but able to use command line when needed.

After a successful training and running "Step1: Evaluate Network", I get the attached error when running "Step2: X-validate". I've tried multiple snapshots of the trained model and get the same "IndexError: index 666 is out of bounds for axis 0 with size 666" for every one of them and don't know what could be the problem. 
![error](https://user-images.githubusercontent.com/62612214/98556608-3aea1400-2271-11eb-9247-f9e664a5c624.PNG)

**Comment by MMathisLab** [comment]
looks like your training dataset perhaps changed?

**Comment by blogeman** [comment]
Thanks for pointing me towards the training dataset. I still couldn't figure out what was going wrong so I just deleted the training dataset, recreated it, and trained the model again. This got rid of the issue.

**Comment by donglabimaging** [comment]
@blogeman Hi, I ran into the same issue. I deleted the latest iteration (iteration-2), recreated it again, and trained the model again.  However, the problem still persist at the x-validation stage.  May I ask if you have deleted the previous iterations as well?  Thank you.

Here is the output:
IndexError: index 2510 is out of bounds for axis 0 with size 2510

**Comment by blogeman** [comment]
Hi @donglabimaging. I didn't delete the previous iterations, if i remember correctly. There I think there was something wrong with my newly labeled images. For you, I am assuming that between iteration-2 and iteration-1 you have added more labeled frames. I would check those labeled frames and see if anything is off. I've found that deleting frames during the labeling process can sometimes cause wonky results (still not sure why).

**Comment by donglabimaging** [comment]
@blogeman Thank you very much for the prompt response.  I appreciate it.   

You are correct that I have added more labeled frames from iteration-1 to iteration-2.  I did not delete frames during the labeling process.  I will check the labeled frames and see what I find.  Thanks again.

**Comment by donglabimaging** [comment]
@blogeman I should have asked this along with the previous message.  I thought I would be able to find the labeled frames.  May I ask how to inspect the labeled frame from iteration-1?  Thanks.

**Comment by blogeman** [comment]
@donglabimaging I don't know of any good/quick way to check those files, but i suspect everything from iteration-1 is fine. Its the new frames that were added that are causing the problem. You could go back to iteration-1 and this time only add half of your newly labeled frames (save the other half for later), then train the network and then X-validate and see if you get an error. If not, that means the problem frame is in the second half of your newly labeled images. I've had to do this once or twice before to track down some problem frames, but I never figured out what went wrong in those frames.

**Comment by backyardbiomech** [comment]
I once had a similar problem and tracked it down to duplicate video entries in my config file. I'm not sure how they got there, bu the fix was easy: delete duplicate video listings, create training data set, train, evaluate.

**Comment by donglabimaging** [comment]
@blogeman @backyardbiomech Thanks for the input!  That sounds straightforward.  I am going to try that!

**Comment by donglabimaging** [comment]
@blogeman @backyardbiomech X-validation worked!  I really appreciate you both.  Many thanks!!

In case someone else encounter the save issue, here are the steps I did:
1. Backed up the iteration-2 folder
2. Switched back to iteration-1 by changing the config file from 2 to 1 (To make sure that I have made the switch correctly, I went to the 'Refine Labels' tab to see it is asking me locate the 'machinelabel-**iter1**.h5')
3. Analyzed a new novel video
4. Refined tracklets and saved
5. Extracted outlier frames
6. Refined labels
7. Merged dataset
7. Created a new data set, and trained again.
